packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -48,6 +48,31 @@ export interface ParametersGatherer<T> {
   gather(): Promise<CancelResponse | ContinueResponse<T>>;
 }
 
+export class CompositeParametersGatherer<T> implements ParametersGatherer<T> {
```

Comment:
We can use `any` for now. I will see how to express this more safely in the type system.